### ISSUE TITLE
Connect via the Cassandra support to Cosmos DB

### ISSUE THREAD

**Issue by synle** [sqlectron/sqlectron-gui#711]
Original Comment from @cicorias

```
Adding to this, has anyone been able to connect via the Cassandra support to Cosmos DB?
```

**Comment by synle** [comment]
Look like @cicorias is referring to the Cassandra setup from within CosmosDB

I haven't tested that scenario, most of my locals are with docker container for Cassandra and as for CosmosDB (Core SQL). I can take a look

![image](https://user-images.githubusercontent.com/3792401/181570057-222bf16b-f362-459f-b87f-6a81619e0357.png)

I think the issue is with auth provider. It's not plain username and password, but rather needs to use `cassandra.auth.PlainTextAuthProvider`

**Comment by cicorias** [comment]
sorry for such a delayed response. It probably is SSL/TLS related, as I can attach with csqlsh passing in the --ssl switch.

I did notice that the SSL option is NOT "enabled" nor can I change it in the UX for the settings.

I did try setting NODE_TLS_REJECT_UNAUTHORIZED in the env prior, still see same issue.

```
Connection Error
Error invoking remote method 'DB_CONNECT': NoHostAvailableError: All host(s) tried for query failed. First host tried, 52.146.79.X:10350: OperationTimedOutError: The host 52.146.79.X:10350 did not reply before timeout 12000 ms at Timeout.requestTimedOut [as _onTimeout] (C:\Users\scicoria\AppData\Local\Programs\sqlectron\resources\app.asar\node_modules\cassandra-driver\lib\operation-state.js:93:28) at listOnTimeout (internal/timers.js:554:17) at processTimers (internal/timers.js:497:7) { info: 'Represents a client-side error that is raised when the client did not hear back from the server within socketOptions.readTimeout', host: '52.146.79.X:10350' }. See innerErrors.
```


<img width="425" alt="image" src="https://user-images.githubusercontent.com/1080406/181771930-ab220dd5-59ea-4096-ae4e-ab57dbf7edb1.png">


<img width="1170" alt="image" src="https://user-images.githubusercontent.com/1080406/181772740-b84e0b55-b56f-4de5-a611-7df19d58fd24.png">

**Comment by synle** [comment]
Hey @cicorias, the issue is with the SSL. I worked on another project called `sqlui-native` (similar to sqlectron - link here https://github.com/synle/sqlui-native). 

Essentially we need to update sqlectron to pass in the SSL Option to support cassandra. Below is the snippet of the code that describes the connection.

```javascript
const client = new cassandra.Client({
  ...clientOptions,
  ...{
    sslOptions: {
      rejectUnauthorized: false,
    }
  }
});
```

It's been a while since I worked on sqlectron, I need to dig in and find out a way to add ssl support for cassandra in sqlectron to make it work with CosmosDB with Cassandra API.
